refinery/ui/source/js/user-file-browser/ctrls/user-file-browser-filters-ctrl.js

```diff
@@ -5,51 +5,22 @@
   .module('refineryUserFileBrowser')
   .controller('UserFileBrowserFiltersCtrl', UserFileBrowserFiltersCtrl);
 
-  function UserFileBrowserFiltersCtrl () {
+  UserFileBrowserFiltersCtrl.$inject = [
+    '$q',
+    'userFileBrowserFactory'
+  ];
+
+  function UserFileBrowserFiltersCtrl ($q, userFileBrowserFactory) {
     var vm = this;
-    vm.attributeFilters = {
-      Technology: {
-        facetObj: [
-          { name: 'ChIP-seq', count: 4 },
-          { name: 'RNA-seq', count: 7 }
-        ]
-      },
-      Organism: {
-        facetObj: [
-          { name: 'Homo sapiens', count: 3 },
-          { name: 'Mus musculus', count: 8 }
-        ]
-      },
-      Filetype: {
-        facetObj: [
-          { name: 'GCT file', count: 42 },
-          { name: 'Affymetrix Probe Results', count: 8 }
-        ]
-      },
-      Owner: {
-        facetObj: [
-          { name: 'Chuck McCallum', count: 6 },
-          { name: 'Geoff Nelson', count: 6 }
-        ]
-      },
-      Antibody: {
-        facetObj: [
-          { name: 'HNF4A', count: 5 },
-          { name: 'FRTS4', count: 46 }
-        ]
-      },
-      Celltype: {
-        facetObj: [
-          { name: 'Caco-2', count: 42 },
-          { name: 'HeLa', count: 1 }
-        ]
-      },
-      Genotype: {
-        facetObj: [
-          { name: 'C57BL/6J', count: 12 }
-        ]
-      }
-    };
+    var promise = $q.defer();
+    var getUserFiles = userFileBrowserFactory.getUserFiles;
+    getUserFiles().then(function (solr) {
+      vm.attributeFilters =
+          userFileBrowserFactory.createFilters(solr.attributes, solr.facet_field_counts);
+      promise.resolve();
+    }, function () {
+      promise.reject();
```

Comment:
Is just `$log.error` sufficient? I don't think there's any real way to recover if the request does fail...